### PR TITLE
[fix] changes for GitHub actions PR CI for matching oneDAL Nightly-build

### PR DESCRIPTION
**Pull request by icfaust**
## Description

Fixes to return CI to green as Nightly-build was changed in oneapi-src/oneDAL#2865 and oneapi-src/oneDAL#2920

No operational changes made, thus benchmarking unnecessary.

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Comment by icfaust** [comment]
Still a problem with kneighbors and stability, this probably will need to be deselected and a ticket made for investigation.

File: deselected_tests.yaml [hunk 1]
```diff
@@ -382,7 +382,10 @@ deselected_tests:
 
   # Deselections for 2025.0
   - ensemble/tests/test_forest.py::test_importances[ExtraTreesRegressor-squared_error-float64]
-
+  - cluster/tests/test_k_means.py::test_kmeans_elkan_results[42-0.01-sparse-normal]
+  - cluster/tests/test_k_means.py::test_kmeans_elkan_results[42-1e-08-sparse-normal]
+  - cluster/tests/test_k_means.py::test_kmeans_elkan_results[42-0-sparse-normal]
+  - cluster/tests/test_k_means.py::test_kmeans_elkan_results[42-1e-100-sparse-normal]
```
Comment (samir-nasibli) [inline deselected_tests.yaml:388]: is this also reproduced on github actions? if not I would suggest revert this, since #2081 targeted to this

File: sklearnex/cluster/tests/test_kmeans.py [hunk 2]
```diff
@@ -128,8 +128,10 @@ def test_results_on_dense_gold_data(dataframe, queue, algorithm):
 def test_dense_vs_sparse(queue, init, algorithm, dims):
     from sklearnex.cluster import KMeans
 
-    if init == "random":
-        pytest.skip("Random initialization in sparse K-means is buggy.")
+    if init in ["random", "k-means++"]:
+        pytest.skip(
+            "Random and K-means++ initialization for sparse K-means are non-conformant."
+        )
```
Comment (samir-nasibli) [inline sklearnex/cluster/tests/test_kmeans.py:134]: If this reproduced only github actions? I guess we can you _IS_INTEL for skip condition

File: sklearnex/tests/utils/base.py [hunk 3]
```diff
@@ -344,3 +346,127 @@ def gen_dataset(
...
+        proc = (
+            subprocess.check_output(["/usr/bin/sysctl", "-n", "machdep.cpu.brand_string"])
+            .strip()
+            .decode("utf-8")
+        )
+
+    return proc
+
+
+_NONINTEL_DESELECTIONS = [
```
Comment (samir-nasibli) [inline sklearnex/tests/utils/base.py:371]: almost all tests for the certain algos disabled I would suggest use condition check for the estimator and disable it in the test suit instead of having this deselections here. Expecting less code for this

File: deselected_tests.yaml (hunk 1 above)
Comment (icfaust) [reply deselected_tests.yaml:388]: Then we can merge his first, this will have a fully green CI, and we can then merge it into this PR.

File: sklearnex/tests/test_run_to_run_stability.py [hunk 4]
```diff
@@ -154,6 +155,8 @@ def test_standard_estimator_stability(estimator, method, dataframe, queue):
         pytest.skip(f"variation observed in {estimator}.score")
     if estimator in ["IncrementalEmpiricalCovariance"] and method == "mahalanobis":
         pytest.skip("allowed fallback to sklearn occurs")
+    if not _IS_INTEL and ("Neighbors" in estimator or "LocalOutlierFactor" in estimator) and method in ["score", "predict", "kneighbors", "kneighbors_graph"]:
+        pytest.skip(f"{estimator} shows instability in Non-Intel hardware")
```
Comment (samir-nasibli) [inline sklearnex/tests/test_run_to_run_stability.py]: I propose to make a common lambda for checking this condition as the nature of these crashes is the same

File: sklearnex/tests/test_run_to_run_stability.py [hunk 5]
```diff
@@ -228,6 +259,14 @@ def test_other_estimator_stability(estimator, method, dataframe, queue):
...
     if "NearestNeighbors" in estimator and "radius" in method:
         pytest.skip(f"RadiusNeighbors estimator not implemented in sklearnex")
+    if (
+        not _IS_INTEL
+        and ("Neighbors" in estimator or "LocalOutlierFactor" in estimator)
+        and method in ["score", "predict", "kneighbors", "kneighbors_graph"]
+    ):
+        if daal_check_version((2025, "P", 200)):
+            pytest.fail("Re-verify failure of algorithms in 2025.2 oneDAL")
+        pytest.skip(f"{estimator} shows instability on Non-Intel hardware")
```
Comment (samir-nasibli) [inline sklearnex/tests/test_run_to_run_stability.py:269]: I propose to make a common function for checking and disabling as the nature of these crashes is the same. 
Also less code duplication

```python
def _skip_neighbors(estimator, estimator):
    if (
        not _IS_INTEL
        and ("Neighbors" in estimator or "LocalOutlierFactor" in estimator)
        and method in ["score", "predict", "kneighbors", "kneighbors_graph"]
    ):
        if daal_check_version((2025, "P", 200)):
            pytest.fail("Re-verify failure of algorithms in 2025.2 oneDAL")
        pytest.skip(f"{estimator} shows instability on Non-Intel hardware")
```

```suggestion
    _skip_neighbors(estimator, estimator)
```

**Review by samir-nasibli — COMMENTED** [review]
@icfaust thank you for the work done! Seems we have green ci now.
#2081 is already merged. Please rebase and address comments. Good to go!

**Review by samir-nasibli — APPROVED** [review]
don't have permissions to Ian's branch. Going to merge this and add refactoring my suggestion as a follow up. Don't want to block green CI enabling
